addons/sales_team/models/crm_team.py

```diff
@@ -224,9 +237,10 @@ def create(self, vals_list):
 
     def write(self, values):
         res = super(CrmTeam, self).write(values)
-        # manually launch company sanity check
-        if values.get('company_id'):
-            self.crm_team_member_ids._check_company(fnames=['crm_team_id'])
+
+        if values.get('company_id'):  # Force re-check of memberships constraint for this team
+            for team in self:
+                team.crm_team_member_ids._constrains_membership()
```

Comment:
Why not calling it in batch directly ?
```suggestion
            self.crm_team_member_ids._constrains_membership()
```

Reply:
done